Model/Product/Builder.php

```diff
@@ -234,8 +236,14 @@ public function build(
             'nosto_product_load_after',
             ['product' => $nostoProduct, 'magentoProduct' => $product]
         );
-
-        return $nostoProduct;
+        $this->eventManager->dispatch(
+            'nosto_product_load_before',
+            ['product' => $nostoProduct, 'magentoProduct' => $product, 'valid' => $valid]
```

Comment:
Rename `valid` to `modelFilter`. This event also needs to be dispatched in the beginning of this method.

Reply:
Ok